### PR TITLE
format test report

### PR DESCRIPTION
**Pull request by ansyral**
Signed-off-by: xuzho <xuzho@microsoft.com>

### CODE REVIEW

**Comment by ansyral** [comment]
It now looks like below and you can choose to see passed/failed tests only.
image

File: extension/src/testResultAnalyzer.ts [hunk 1]
```diff
@@ -106,7 +106,7 @@ export class TestResultAnalyzer {
 
         t.result = {
             status: notRun ? undefined : (skipNum === t.children.length ? TestStatus.Skipped : (failNum > 0 ? TestStatus.Fail : TestStatus.Pass)),
-            summary: `Total tests run: ${passNum + failNum}, Failures: ${failNum}, Skips: ${skipNum}.`,
+            summary: `Total tests: ${t.children.length}. Runs: ${passNum + failNum}, Failures: ${failNum}, Skips: ${skipNum}.`,
```
Comment (akaroml) [inline extension/src/testResultAnalyzer.ts]: `Tests run: 47, Failures: 0, Errors: 0, Skipped: 0`
This is the output from Maven command line. Please use this format.

File: extension/src/extension.ts [hunk 2]
```diff
@@ -221,7 +221,7 @@ async function runSingleton(javaHome: string, tests: TestSuite[] | TestSuite, st
 function showDetails(test: TestSuite) {
     const editor = window.activeTextEditor;
     const uri: Uri = encodeTestSuite(editor.document.uri, test);
-    return workspace.openTextDocument(uri).then((doc) => window.showTextDocument(doc, editor.viewColumn + 1));
+    return commands.executeCommand('vscode.previewHtml', uri, ViewColumn.Two, 'Test Report Preview');
```
Comment (akaroml) [inline extension/src/extension.ts]: Test Report Preview -> Test Report

File: extension/src/extension.ts [hunk 3]
```diff
@@ -221,7 +221,7 @@ async function runSingleton(javaHome: string, tests: TestSuite[] | TestSuite, st
 function showDetails(test: TestSuite) {
     const editor = window.activeTextEditor;
     const uri: Uri = encodeTestSuite(editor.document.uri, test);
```
Comment (akaroml) [inline extension/src/extension.ts]: Why do we have encodeTestSuite and decodeTestSuite?
The Uri lib can handle this perfectly.

File: extension/src/testReportProvider.ts [hunk 4]
```diff
@@ -1,51 +1,180 @@
...
-        report += JSON.stringify(test.result, null, 4);
-        if (test.level === TestLevel.Method) {
-            return report;
-        }
-        report += os.EOL;
-        for (const child of test.children) {
-            report += this.getTestReport(child) + os.EOL;
+    private errorSnippet(error: string): string {
+        return `
+            <body>
```
Comment (akaroml) [inline extension/src/testReportProvider.ts]: Hard-coding HTML is never good. Please use a template engine. In this way, the HTML can be extracted as a separate template file and be maintained separately.

File: extension/src/testReportProvider.ts [hunk 5]
```diff
@@ -1,51 +1,180 @@
...
+            case TestLevel.Method:
+                return this.methodSnippet(test);
+            default:
+                return this.errorSnippet('Not supported test level. Currently support class level and method level.');
         }
-        return report;
+    }
+
+    private classSnippet(test: TestSuite): string {
+        const totalCount: number = test.children.length;
```
Comment (akaroml) [inline extension/src/testReportProvider.ts]: Use template engine.

File: extension/src/extension.ts (hunk 3 above)
Comment (ansyral) [reply extension/src/extension.ts]: we need to add query string to the uri

**Comment by ansyral** [comment]
close this one as it is split to several smaller reviews

**Pull request closed without merging**